packages/story-editor/src/app/rightClickMenu/hooks/useIsPartiallyOffCanvas.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  DANGER_ZONE_HEIGHT,
+  PAGE_HEIGHT,
+  PAGE_WIDTH,
+} from '@googleforcreators/units';
+import { useCallback } from 'react';
+
+function useIsPartiallyOffCanvas(selectedElement) {
```

Comment:
Would be nice to cater for element rotation here as well. Maybe in a follow-up, but I think we can avoid using custom checks for each side of the element here. Basically calculating the area of two overlapping rectangles.

Adding tests for this functionality would be an excellent way to prevent regressions when doing such a migration.

Reply:
Added: https://github.com/GoogleForCreators/web-stories-wp/issues/12203 to cover element rotation.

Also added unit tests for the current off-screen detection.